### PR TITLE
Reduces travis run time.

### PR DESCRIPTION
**Pull request by sileht**

### CODE REVIEW

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 70f481d91cea43007f6533924781a8190cd488c8

**Comment by sileht** [comment]
Pull-request updated, HEAD is now e3231416b4cd850f44a03efcb4c6cfd77feffd08

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 3d499bdb3f8a5e71498dcc5d1930c301921616c7

**Comment by sileht** [comment]
Pull-request updated, HEAD is now b2e56c6cd3fa52376149aed8f02cc717206ed669

**Comment by sileht** [comment]
Pull-request updated, HEAD is now fc0b43a068b3fc44a3871c2fb4c57167e0ba0043

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 68272d9cedf9ad4006f5d5c29ad650049a10c355

**Comment by sileht** [comment]
Pull-request updated, HEAD is now ca37f904490929234276ba3bfbfbb34ccda00508

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 40d9a4d1554c2a7f22f8b1d02ef86f6e79b9d69e

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 37cda643e8a3236e38e6ea6d31c4fb6e67a1c921

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 8a5eb1eaf64917968318e3dfc2fe13b7c514b2e4

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 37cda643e8a3236e38e6ea6d31c4fb6e67a1c921

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 041c8714f4cf5945dff6f02a40b881c512b516a6

File: tox.ini [hunk 1]
```diff
@@ -26,16 +26,18 @@ setenv =
...
     GNOCCHI_TEST_TARBALLS=http://tarballs.openstack.org/swift/swift-master.tar.gz#egg=swift
     ceph: GNOCCHI_TEST_TARBALLS=
     swift: GNOCCHI_TEST_TARBALLS=http://tarballs.openstack.org/swift/swift-master.tar.gz#egg=swift
     s3: GNOCCHI_TEST_TARBALLS=
     redis: GNOCCHI_TEST_TARBALLS=
     file: GNOCCHI_TEST_TARBALLS=
-deps = .[test]
-   postgresql: .[postgresql,{env:GNOCCHI_STORAGE_DEPS}]
-   mysql: .[mysql,{env:GNOCCHI_STORAGE_DEPS}]
+deps = .[test,{env:GNOCCHI_STORAGE_DEPS:},{env:GNOCCHI_INDEXER_DEPS:}]
```
Comment (chungg) [inline tox.ini]: env:GNOCCHI_STORAGE_DEPS: what's the trailing ':' for?

File: .travis.yml [hunk 2]
```diff
@@ -8,31 +7,20 @@ cache:
...
-  - TARGET: docs-gnocchi.xyz
-
-  - TARGET: py27-mysql-ceph-upgrade-from-3.1
-  - TARGET: py35-postgresql-file-upgrade-from-3.1
-
-  - TARGET: py27-mysql
-  - TARGET: py35-mysql
-  - TARGET: py27-postgresql
-  - TARGET: py35-postgresql
+  - TARGETS: pep8,docs,docs-gnocchi.xyz,py27-mysql-ceph-upgrade-from-3.1,py35-postgresql-file-upgrade-from-3.1
```
Comment (chungg) [inline .travis.yml]: does this mean upgrade,docs, and pep8 all run on same container and py27(postgresql+mysql) and py35(postgresql+mysql) run on another docker image

Comment (jd) [inline .travis.yml]: Yes.

The downside is that this one hell of a big log for all those jobs, and we need to restart all of them if just one fails.

I wonder if it's really worth it.

**Comment by sileht** [comment]
Pull-request updated, HEAD is now b5499088401f8b9f4989576be6558dbb607fa40c

File: .travis.yml (hunk 2 above)
Comment (chungg) [inline .travis.yml]: hmm... yeah, those pyxx builds are going to be massive... and sequential? our are jobs right now only one by one as well?

i'd prefer if we split the pyxx jobs but if it's faster than i can live with massive logs.

File: tox.ini (hunk 1 above)
Comment (sileht) [inline tox.ini]: what comes after the : is the default

File: .travis.yml (hunk 2 above)
Comment (sileht) [inline .travis.yml]: The more problematic issue is the pyXX-<indexer> jobs that take 30-40 minutes each, other are really fast.

The gain of serializing many targets in one container, it to not download each time the 500m of the ubuntu image.

I will try another PR with only the run-tests.sh hack to see.

**Comment by sileht** [comment]
Recent change have already reduce the times from ~ 45 minutes accumulated.

**Pull request closed without merging**